### ISSUE TITLE
How to tune the loss weights?

### ISSUE THREAD

**Issue by twmht** [yzd-v/FGD#11]
Hi,

there are some of the loss weights, like https://github.com/yzd-v/FGD/blob/master/mmdet/distillation/losses/fgd.py#L27

How did you tune these loss weights? Any guideline to tune these hyper parameters?

**Comment by yzd-v** [comment]
It's used to balance the losses and FGD is not very sensitive to them. You can change the hyper-parameters according to the loss scale.

**Comment by twmht** [comment]
what did you mean  `loss scale`? Is it related to classification loss and regression loss?

**Comment by yzd-v** [comment]
The smallest distillation loss scale is similar to the original classification loss or regression loss may be fine.

**Comment by yzd-v** [comment]
yes, 0.1-2 may be fine too

**Comment by twmht** [comment]
I found it may be sensitive to these parameters in my own dataset. 

I would follow your suggestion and see whether it works or not.

**Comment by yzd-v** [comment]
You can keep the ratio between different loss and change them together to balance loss. You may also check whether the teacher load the weight successfully.

**Comment by twmht** [comment]
@yzd-v 

Did you mind to upload your training log?

**Comment by twmht** [comment]
Here are my distiller loss

Are these losses too small?

![image](https://user-images.githubusercontent.com/1567200/158732516-a4acbb2b-5a06-4681-97d3-c3cb940ddedd.png)

**Comment by twmht** [comment]
update 

![image](https://user-images.githubusercontent.com/1567200/158740978-6c976b01-f0d6-461e-9487-195271e0f4e2.png)

**Comment by yzd-v** [comment]
There might be something wrong in your distillation environment. You can test your environment on COCO to compare the results.

**Comment by twmht** [comment]
yup I am going to test with coco.

**Comment by luowei0701** [comment]
> The smallest distillation loss scale is similar to the original classification loss or regression loss may be fine.

but，in my training, fpn0 and fpn4 scales differ by a factor of 200, what should I do？

**Comment by renyu-renyu** [comment]
Do you use FGD to distill your own data?

**Comment by renyu-renyu** [comment]
I am researching this direction, can I exchange some questions about FGD with you?

**Comment by twmht** [comment]
@renyu-renyu 

Are you talking with me?

**Comment by renyu-renyu** [comment]
> @renyu-renyu
> 
> 你在跟我说话吗？

Yes, if we can, we can communicate together

**Comment by twmht** [comment]
Yup.

What is your opinion about FGD?

**Comment by renyu-renyu** [comment]
> 是的。
> 
> 您对 FGD 有什么看法？

Are you training your own dataset or?

**Comment by twmht** [comment]
Yup.

I was training widerface.

but did not get good result, what about you?

**Comment by renyu-renyu** [comment]
> 是的。
> 
> 我正在训练更宽的脸。
> 
> 但没有得到好的结果，你呢？

I'm reproducing the results of FGD and haven't trained my own network and data

**Comment by renyu-renyu** [comment]
> 

I'm reproducing the results of FGD and haven't trained my own network and data

**Comment by twmht** [comment]
did you reproduce?

**Comment by renyu-renyu** [comment]
> 你繁殖了吗？
It is reproduced, but the accuracy is not ideal, and it may need to be modified. Have you implemented your own data set?

**Comment by twmht** [comment]
what do you mean reproduced but the accuracy is not ideal? does the accuracy dot match the paper?

**Comment by renyu-renyu** [comment]
> 你是什​​么意思复制但准确性不理想？精度点是否与纸张匹配？

yes

**Comment by twmht** [comment]
Can you share your reproduced accuracies and the paper accuracies?

**Comment by renyu-renyu** [comment]
> 你能分享你的复制精度和纸张精度吗？

I'm not quite done yet, I've emailed you

**Comment by dd1005** [comment]
> Can you share your reproduced accuracies and the paper accuracies?

Will you reproduce this article? How about the accuracy of reproduction? I'd like to discuss it with you

**Comment by dd1005** [comment]
> > 你能分享你的复制精度和纸张精度吗？
> 
> I'm not quite done yet, I've emailed you

Will you reproduce this article? How about the accuracy of reproduction? I'd like to discuss it with you

**Comment by dd1005** [comment]
> > 你能分享你的复制精度和纸张精度吗？
> 
> I'm not quite done yet, I've emailed you

Will you reproduce this article? How about the accuracy of reproduction? I'd like to discuss it with you

**Comment by peiyingxin** [comment]
> > > 你能分享你的复制精度和纸张精度吗？
> > 
> > 
> > I'm not quite done yet, I've emailed you
> 
> Will you reproduce this article? How about the accuracy of reproduction? I'd like to discuss it with you

Have you reproduce this article? I'd like to discuss it with you.
